scripts/linking.py

```diff
@@ -463,9 +465,13 @@ def eval_candidate_generation_and_linking(examples: List[data_util.MedMentionExa
                     for candidate_id in candidate_ids:
                         has_definition = 'definition' in umls_concept_dict_by_id[candidate_id]
                         cosine_scores = [cosine for alias, cosine in candidates[candidate_id]]
-                        classifier_example = ({'has_definition': has_definition, 'cosines': cosine_scores,
+                        label = int(gold_entity.umls_id == candidate_id)
```

Comment:
Please can you pull this out into its own function,

`generate_linking_classifier_training_data`. It has some small amount of overlapping code but it would be much cleaner and easier to read.

Reply:
I doubt the code overlap is going to be small, but the other issue is that with every change in the candidate generation here, we will need to update the other function as well. That's why I think it is better to keep it here. Also, with the code refactoring, it looks a lot better.